wordpress-mobile/WordPress-Android
# Issue/9666 post list show real progress

Fixes #9666 

This PR is follow up for draft PR https://github.com/wordpress-mobile/WordPress-Android/pull/9708 which I've closed to make things clearer.

**Note: This PR is based on https://github.com/wordpress-mobile/WordPress-Android/pull/9770.**
Merge instructions

0. Review this PR
1. Wait until https://github.com/wordpress-mobile/WordPress-Android/pull/9770 is merged
2. Update target branch to `feature/post-filters-compat-view-toggle`
3. (Review) and Merge this PR


Adds a determinate progress bar indicator to post list items when media files are being uploaded.

To test:

1. Open Post List
2. Create a post
3. Add title and add some images from the local storage
4. Publish the post before the images are uploaded (you might want to modify network settings on your emulator to make the internet connection slower)
5. Notice the progress bar on the item in the list is updated as the upload progresses

Update release notes:

- I haven't updated the release notes as this issue will be released along with the Post Filters project
![real-progress](https://user-images.githubusercontent.com/2261188/57069667-90854e00-6cd5-11e9-9edb-e276f86f16ce.gif)
